ext/opentelemetry-ext-pymongo/src/opentelemetry/ext/pymongo/__init__.py

```diff
@@ -0,0 +1,78 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+The opentelemetry-ext-pymongo package allows tracing commands made by the
+pymongo library.
+"""
+
+from pymongo import monitoring
+from opentelemetry.trace import SpanKind, Span
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+DATA_BASE_TYPE = "mongodb"
+COMMAND_ATTRIBUTES = ["filter", "sort", "skip", "limit", "pipeline"]
+
+
+def trace_integration(tracer=None):
+    """Integrate with pymongo to trace it using event listener.
+       https://api.mongodb.com/python/current/api/pymongo/monitoring.html
+    """
+    monitoring.register(CommandTracer(tracer))
+
+
+class CommandTracer(monitoring.CommandListener):
+    def __init__(self, tracer):
+        self._tracer = tracer
+
+    def started(self, event: monitoring.CommandStartedEvent):
+        name = (
+            DATA_BASE_TYPE
+            + "."
+            + event.command_name
+            + "."
+            + event.command.get(event.command_name)
+        )
+        span = self._tracer.start_span(name, kind=SpanKind.CLIENT)
+        span.set_attribute("component", DATA_BASE_TYPE)
+        span.set_attribute("db.type", DATA_BASE_TYPE)
+        span.set_attribute("db.instance", event.database_name)
+        span.set_attribute(
+            "db.statement",
+            event.command_name + " " + event.command.get(event.command_name),
+        )
+        if event.connection_id is not None:
+            span.set_attribute("peer.address", str(event.connection_id))
```

Comment:
Looking at the test it seems that this will lead to a Python tuple repr. Since we already have hostname and port, I think we should leave the address out unless we find a better, URL-like property on the event.

Reply:
The spec have address as a required field, that is the reason I left it in there

https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md#databases-client-calls

Reply:
The spec also says that it must be a "JDBC substring". 🤷‍♂  I guess the DB conventions are not really implementable here.

Reply:
You have a point there I will remove it, we can revisit later if spec is updated